pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -37,14 +43,41 @@ const (
 	clusterDeploymentAdmissionGroup    = "admission.hive.openshift.io"
 	clusterDeploymentAdmissionVersion  = "v1alpha1"
 	clusterDeploymentAdmissionResource = "clusterdeployments"
+
+	// ManagedDomainsFileEnvVar if present, points to a simple text
+	// file that includes a valid managed domain per line. Cluster deployments
+	// requesting that their domains be managed must have a base domain
+	// that is a direct child of one of the valid domains.
+	ManagedDomainsFileEnvVar = "MANAGED_DOMAINS_FILE"
```

Comment:
Would it make sense to have a general purpose config file for hive admission that could contain "validDomains" as a section?

I'm thinking that in the future, it might be nice to have a place to put other config values and having a structure in place would be really nice.

Reply:
@twiest would it be ok to add that file when we need it? I don't think this approach limits our choices going forward.

Reply:
@csrwng I was just thinking it might be nice to have 1 config file and have the infra already built out for the next one.

It's honestly fine this way.